crypto/equihash.cpp

```diff
@@ -42,28 +42,56 @@ int Equihash<N,K>::InitialiseState(eh_HashState& base_state, const char* persona
     memcpy(personalization+12, &le_K, 4);
     return crypto_generichash_blake2b_init_salt_personal(&base_state,
                                                          NULL, 0, // No key.
-                                                         (512/N)*N/8,
+                                                         (512/N)*((N+7)/8),
```

Comment:
Will changing these cause issues for other chains? There are several lines below where lines like this are modified also.

Reply:
Funnily that was my first question to lollidieb also when merging this code ;-)

that should be no problem... before the assumption was that N is divisible by 8 Now It stays the same if divisible, if not round it (required by 125,4).